File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/OacObjectColumnSqlInfo.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * SQL metadata for a column in an OAC data object.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=OacObjectColumnSqlInfo.Builder.class)

public final class OacObjectColumnSqlInfo  {
    @Deprecated
    @java.beans.ConstructorProperties({"aggRule", "dataType", "aggType"})
    public OacObjectColumnSqlInfo(String aggRule, String dataType, String aggType) {
        super();
        this.aggRule = aggRule;
        this.dataType = dataType;
        this.aggType = aggType;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The aggregation rule for the column in the OAC data object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("aggRule")
private String aggRule;

        /**
         * The aggregation rule for the column in the OAC data object.
         * @param aggRule the value to set
         * @return this builder
         **/
        

public Builder aggRule(String aggRule) {
    this.aggRule = aggRule;
    return this;
}
            /**
     * The data type for the column in the OAC data object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("dataType")
private String dataType;

        /**
         * The data type for the column in the OAC data object.
         * @param dataType the value to set
         * @return this builder
         **/
        

public Builder dataType(String dataType) {
    this.dataType = dataType;
    return this;
}
            /**
     * The aggregation type for the column in the OAC data object.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("aggType")
private String aggType;

        /**
         * The aggregation type for the column in the OAC data object.
         * @param aggType the value to set
         * @return this builder
         **/
        

public Builder aggType(String aggType) {
    this.aggType = aggType;
    return this;
}


        public OacObjectColumnSqlInfo build() {
            OacObjectColumnSqlInfo model = new OacObjectColumnSqlInfo(this.aggRule
                , this.dataType
                , this.aggType);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(OacObjectColumnSqlInfo model) {
                this.aggRule(model.getAggRule());
    this.dataType(model.getDataType());
    this.aggType(model.getAggType());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The aggregation rule for the column in the OAC data object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("aggRule")
    private final String aggRule;

        /**
     * The aggregation rule for the column in the OAC data object.
     * @return the value
     **/
    
    public String getAggRule() {
        return aggRule;
    }


        /**
     * The data type for the column in the OAC data object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("dataType")
    private final String dataType;

        /**
     * The data type for the column in the OAC data object.
     * @return the value
     **/
    
    public String getDataType() {
        return dataType;
    }


        /**
     * The aggregation type for the column in the OAC data object.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("aggType")
    private final String aggType;

        /**
     * The aggregation type for the column in the OAC data object.
     * @return the value
     **/
    
    public String getAggType() {
        return aggType;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("OacObjectColumnSqlInfo(");
        sb.append("aggRule=").append(String.valueOf(this.aggRule));
        sb.append(", dataType=").append(String.valueOf(this.dataType));
        sb.append(", aggType=").append(String.valueOf(this.aggType));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof OacObjectColumnSqlInfo)) {
            return false;
        }

        OacObjectColumnSqlInfo other = (OacObjectColumnSqlInfo) o;
        return java.util.Objects.equals(this.aggRule, other.aggRule) &&
            java.util.Objects.equals(this.dataType, other.dataType) &&
            java.util.Objects.equals(this.aggType, other.aggType);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.aggRule == null ? 43 : this.aggRule.hashCode());
        result = (result * PRIME) + (this.dataType == null ? 43 : this.dataType.hashCode());
        result = (result * PRIME) + (this.aggType == null ? 43 : this.aggType.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/oac_object_column_sql_info.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class OacObjectColumnSqlInfo(object):
    """
    SQL metadata for a column in an OAC data object.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new OacObjectColumnSqlInfo object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param agg_rule:
            The value to assign to the agg_rule property of this OacObjectColumnSqlInfo.
        :type agg_rule: str

        :param data_type:
            The value to assign to the data_type property of this OacObjectColumnSqlInfo.
        :type data_type: str

        :param agg_type:
            The value to assign to the agg_type property of this OacObjectColumnSqlInfo.
        :type agg_type: str

        """
        self.swagger_types = {
            'agg_rule': 'str',
            'data_type': 'str',
            'agg_type': 'str'
        }

        self.attribute_map = {
            'agg_rule': 'aggRule',
            'data_type': 'dataType',
            'agg_type': 'aggType'
        }

        self._agg_rule = None
        self._data_type = None
        self._agg_type = None

    @property
    def agg_rule(self):
        """
        Gets the agg_rule of this OacObjectColumnSqlInfo.
        The aggregation rule for the column in the OAC data object.


        :return: The agg_rule of this OacObjectColumnSqlInfo.
        :rtype: str
        """
        return self._agg_rule

    @agg_rule.setter
    def agg_rule(self, agg_rule):
        """
        Sets the agg_rule of this OacObjectColumnSqlInfo.
        The aggregation rule for the column in the OAC data object.


        :param agg_rule: The agg_rule of this OacObjectColumnSqlInfo.
        :type: str
        """
        self._agg_rule = agg_rule

    @property
    def data_type(self):
        """
        Gets the data_type of this OacObjectColumnSqlInfo.
        The data type for the column in the OAC data object.


        :return: The data_type of this OacObjectColumnSqlInfo.
        :rtype: str
        """
        return self._data_type

    @data_type.setter
    def data_type(self, data_type):
        """
        Sets the data_type of this OacObjectColumnSqlInfo.
        The data type for the column in the OAC data object.


        :param data_type: The data_type of this OacObjectColumnSqlInfo.
        :type: str
        """
        self._data_type = data_type

    @property
    def agg_type(self):
        """
        Gets the agg_type of this OacObjectColumnSqlInfo.
        The aggregation type for the column in the OAC data object.


        :return: The agg_type of this OacObjectColumnSqlInfo.
        :rtype: str
        """
        return self._agg_type

    @agg_type.setter
    def agg_type(self, agg_type):
        """
        Sets the agg_type of this OacObjectColumnSqlInfo.
        The aggregation type for the column in the OAC data object.


        :param agg_type: The agg_type of this OacObjectColumnSqlInfo.
        :type: str
        """
        self._agg_type = agg_type

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
